src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -117,22 +152,26 @@ internal static string RemoveExtendedPrefix(string path)
             if (IsExtendedUnc(path))
                 return path.Remove(2, 6);
 
-            return path.Substring(4);
+            return path.Remove(0, 4);
         }
 
         /// <summary>
-        /// Removes the extended path prefix (\\?\) if present.
+        /// Remove the extended prefix from the given path. Return the type of path found.
         /// </summary>
-        internal static StringBuilder RemoveExtendedPrefix(StringBuilder path)
+        internal static ExtendedType RemoveExtendedPrefix(ref string path)
```

Comment:
Here is the example:

```C#
        static void Test()
        {
            string foo = "foo";
            Sample.DoSomething(ref foo);
        }

        static void TestToo()
        {
            bool dontCare;
            string foo = "foo";
            foo = Sample.DoSomethingToo(foo, out dontCare);
        }

        public static bool DoSomething(ref string value)
        {
            value.ToUpper();
            return value.Length > 5;
        }

        public static string DoSomethingToo(string value, out bool result)
        {
            result = value.Length > 5;
            return value.ToUpper();
        }
```

You end up with the following in IL from the callers:
```
.method private hidebysig static void  Test() cil managed
{
  // Code size       15 (0xf)
  .maxstack  1
  .locals init ([0] string foo)
  IL_0000:  ldstr      "foo"
  IL_0005:  stloc.0
  IL_0006:  ldloca.s   foo
  IL_0008:  call       bool ConsoleApplication2.Sample::DoSomething(string&)
  IL_000d:  pop
  IL_000e:  ret
}

.method private hidebysig static void  TestToo() cil managed
{
  // Code size       14 (0xe)
  .maxstack  2
  .locals init ([0] bool dontCare)
  IL_0000:  ldstr      "foo"
  IL_0005:  ldloca.s   dontCare
  IL_0007:  call       string ConsoleApplication2.Sample::DoSomethingToo(string,
                                                                         bool&)
  IL_000c:  pop
  IL_000d:  ret
}
```
The maxstack+1 is what I was referring to.